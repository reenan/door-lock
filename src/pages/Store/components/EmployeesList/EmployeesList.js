import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Icon } from 'semantic-ui-react'

import './EmployeesList.scss'
class EmployeesList extends Component {
  render() {
    const {
      employees,
      employeesKeys,
      selectedEmployee,
      selectEmployee,
      roles
    } = this.props

    return (
      <div className='employees'>
        {
          employeesKeys.map((employeeID) => (
            <Employee
              key={employeeID}
              employeeID={employeeID}
              selected={employeeID === selectedEmployee}
              employees={employees}
              roles={roles}
              selectEmployee={selectEmployee}
            />
          ))
        }
      </div>
    )
  }
}

const Employee = ({ employeeID, selected, selectEmployee, employees, roles }) => (
  <div key={employeeID} className={ selected ? 'selected' : ''}
    onClick={selectEmployee.bind(null, employeeID)}>
    <p>
      <Icon className={`custom-icon ${selected ? 'check-circle' : 'circle'}`} />
      <span>{employees[employeeID].name}</span>
      
      <span>
        ({
          employees[employeeID].role ?
          roles[employees[employeeID].role].name : 'No role defined'
        })
      </span>

    </p>
  </div>
)

EmployeesList.propTypes = {
  employeesKeys: PropTypes.array.isRequired,
  employees: PropTypes.object.isRequired,
  roles: PropTypes.object.isRequired,
  selectEmployee: PropTypes.func.isRequired,
  selectedEmployee: PropTypes.string,
}

EmployeesList.defaultProps = {
  selectedEmployee: null,
}

export default EmployeesList
