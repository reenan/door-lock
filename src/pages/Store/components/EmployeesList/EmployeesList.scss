@import '../../../../resources/_mixins.scss';

.employees {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  grid-gap: 15px;
  margin: 30px 0;

  > div {
    @include transition(color .4s,border-color .4s,opacity .4s);
    cursor: pointer;
    border: 1px solid #fff;
    border-radius: 5px;
    padding: 15px;
    opacity: .7;
    background-color: #0000001a;

    p > span + span {
      margin-left: 5px;
    }

    &:hover, &.selected {
      opacity: 1;
      color: #25cb68;
      border-color: #25cb68;
    }
  }
}