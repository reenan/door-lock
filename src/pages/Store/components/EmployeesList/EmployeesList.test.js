import React from 'react'
import EmployeesList from './EmployeesList'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <EmployeesList
      employeesKeys={[]}
      employees={{}}
      roles={{}}
      selectEmployee={() => {}}
      selectedEmployee={null}
    />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with employees', () => {
  const component = renderer.create(
    <EmployeesList
      employeesKeys={['mock']}
      employees={{
        mock: {
          name: 'Mock',
          role: 'mock'
        }
      }}
      roles={{
        mock: {
          name: 'Mock'
        }
      }}
      selectEmployee={() => {}}
      selectedEmployee={null}
    />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with employees and selected employee', () => {
  const component = renderer.create(
    <EmployeesList
      employeesKeys={['mock']}
      employees={{
        mock: {
          name: 'Mock',
          role: 'mock'
        }
      }}
      roles={{
        mock: {
          name: 'Mock'
        }
      }}
      selectEmployee={() => {}}
      selectedEmployee='mock'
    />
  )

  expect(component).toMatchSnapshot()
})
