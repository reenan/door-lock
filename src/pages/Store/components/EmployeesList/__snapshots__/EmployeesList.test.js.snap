// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="employees"
/>
`;

exports[`renders without crashing with employees 1`] = `
<div
  className="employees"
>
  <div
    className=""
    onClick={[Function]}
  >
    <p>
      <i
        aria-hidden="true"
        className="icon custom-icon circle"
        onClick={[Function]}
      />
      <span>
        Mock
      </span>
      <span>
        (
        Mock
        )
      </span>
    </p>
  </div>
</div>
`;

exports[`renders without crashing with employees and selected employee 1`] = `
<div
  className="employees"
>
  <div
    className="selected"
    onClick={[Function]}
  >
    <p>
      <i
        aria-hidden="true"
        className="icon custom-icon check-circle"
        onClick={[Function]}
      />
      <span>
        Mock
      </span>
      <span>
        (
        Mock
        )
      </span>
    </p>
  </div>
</div>
`;
